client/my-sites/sidebar/utils.js

```diff
@@ -5,18 +5,22 @@ import { includes, some } from 'lodash';
 
 // Check if `path` starts with `pathPrefix` and the very next character
 // (if there is any) is `/` or `?`
-function pathStartsWith( fullPath, pathPrefix ) {
+// If shallow is true, check only the very first fragment.
+function pathStartsWith( fullPath, pathPrefix, shallow ) {
```

Comment:
I'd be tempted to make `shallow` part of an options object. Passing a bool as an arg to a function makes it difficult to understand what's happening without finding and reading the func signature.

Reply:
I have opted in creating separate function for our unified sidebar.